### PR TITLE
feat: add --node-env flag

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
YES WIP
**If relevant, did you update the documentation?**
YES
**Summary**
fixes https://github.com/webpack/webpack-cli/issues/2362

**Does this PR introduce a breaking change?**
NO

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/webpack-cli.js [hunk 1]
```diff
@@ -1359,6 +1365,10 @@ class WebpackCLI {
             process.exit(2);
         }
 
+        if (typeof options.nodeEnv === 'string') {
```
Comment (alexander-akait) [inline packages/webpack-cli/lib/webpack-cli.js:1368]: Let' create new method `applyNodeEnv` and run it here https://github.com/webpack/webpack-cli/blob/bcf1c2c0f76642d9bfc63a78f3bac3cd324b5547/packages/webpack-cli/lib/webpack-cli.js#L1656 to avoid mixing logic

**Review by alexander-akait — COMMENTED** [review]
Also need tests, anyway looks good

File: test/node-env/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    mode: 'development',
+};
```
Comment (alexander-akait) [inline test/node-env/webpack.config.js:3]: Just add plugin to output options and use `mode: process.env.NODE_ENV`

File: test/node-env/node-env.test.js [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
-//});
+describe('--node-env flag', () => {
+    it('should set process.env.NODE_ENV', () => {
+        const { exitCode, stderr, stdout } = run(__dirname, ['--node-env', 'production']);
+
+        expect(exitCode).toBe(0);
+        expect(stderr).toBeFalsy();
+        expect(stdout).toContain("mode: 'production'");
+    });
+});
```
Comment (alexander-akait) [inline test/node-env/node-env.test.js:13]: Let's add test for `development` and `none` (as webpack supported)

**Comment by alexander-akait** [comment]
@snitin315 let's finish, I want to do release, it is very high requested feature

**Comment by snitin315** [comment]
Need to fix CI, debugging.

File: test/node-env/node-env.test.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    it('should set process.env.NODE_ENV to "production"', () => {
+        const { exitCode, stderr, stdout } = run(__dirname, ['--node-env', 'production']);
+
+        expect(exitCode).toBe(0);
+        expect(stderr).toBeFalsy();
+        expect(stdout).toContain("mode: 'production'");
+    });
+
+    it('should set process.env.NODE_ENV to "development"', () => {
+        const { exitCode, stderr, stdout } = run(__dirname, ['--node-env', 'production']);
```
Comment (alexander-akait) [inline test/node-env/node-env.test.js:15]: I think here mistake `--node-env development`

File: test/node-env/node-env.test.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+        expect(exitCode).toBe(0);
+        expect(stderr).toBeFalsy();
+        expect(stdout).toContain("mode: 'development'");
+    });
+
+    it('should set process.env.NODE_ENV to "none"', () => {
+        const { exitCode, stderr, stdout } = run(__dirname, ['--node-env', 'production']);
```
Comment (alexander-akait) [inline test/node-env/node-env.test.js:23]: I think here mistake `--node-env none`

**Review by alexander-akait — DISMISSED** [review]
Let's convert from draft

**Comment by alexander-akait** [comment]
I will add some complex tests

**Comment by snitin315** [comment]
💯

**Comment by alexander-akait** [comment]
Thanks
